server/src/main/java/io/crate/metadata/upgrade/MetadataIndexUpgrader.java

```diff
@@ -97,6 +107,80 @@ MappingMetadata createUpdatedIndexMetadata(MappingMetadata mappingMetadata, Stri
         }
     }
 
+    /**
+     * Migrates from "indexed column refers to FT indices via "copy_to"
+     * to "FT index refers to source columns via new "sources" field.
+     *
+     * FULLTEXT indices are defined on CREATE TABLE and can only be a part of top-level "properties".
+     * Hence,we will be adding sources only into fullMappingProperties.
+     *
+     * @param fullMappingProperties is the top level properties of the whole table.
+     * It's propagated (and mutated) through all calls since we might want to access upper level map
+     * when 'copy_to' is part of a deep sub-column.
+     *
+     * @param currentPath is accumulated FQN. On every call we know only leaf name and 'sources' have to contain FQN to be able to resolve Reference.
+     */
+    public static boolean addFulltextSources(Map<String, Object> fullMappingProperties, Map<String, Object> currentLevelMapping, String currentPath) {
+        Map<String, Map<String, Object>> propertiesMap = Maps.get(currentLevelMapping, "properties");
+        if (propertiesMap == null) {
+            return false;
+        }
+        boolean updated = false;
+        for (Map.Entry<String, Map<String, Object>> entry: propertiesMap.entrySet()) {
+            String columnName = entry.getKey(); // Not an FQN name, leaf name.
+            String columnFQN = currentPath.isEmpty() ? columnName : currentPath + "." + columnName;
+            Map<String, Object> columnProperties = entry.getValue();
+
+            // Update FT indices
+            List<String> indices = Maps.get(columnProperties, "copy_to");
+            if (indices != null) {
+                updated = true;
+                for (String ftIndex: indices) {
+                    Map<String, Object> ftProps = Maps.get(fullMappingProperties, ftIndex);
+                    List<String> sources = Maps.get(ftProps, "sources");
+                    if (sources == null) {
+                        sources = new ArrayList<>();
+                        ftProps.put("sources", sources);
+                    }
+                    sources.add(columnFQN);
+                }
+                columnProperties.remove("copy_to");
```

Comment:
As discussed, removing it completely, Upgraders take care of the mixed clusters (?)

Reply:
I think this is the reason of crate-qa failure